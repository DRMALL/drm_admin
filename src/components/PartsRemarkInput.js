

import React from 'react'
import Input from 'antd/lib/input'
import Select from 'antd/lib/select'
import getInputValue from '../actions/getInputValue'
import dispatch from '../actions/dispatch'
import store from '../commons/store'

const PartsRemarkInput = ({machineArray}) => (
  <div className='part-remark-input' >
    <div className='part-remark-input-flex' >
      <div className='part-remark-input-text' >设备编号</div>
      <SelectOption arr={ machineArray } keys='number' />
    </div>
    <div className='part-remark-input-flex' >
      <div className='part-remark-input-text' >设备名称</div>
      <SelectOption arr={ machineArray } keys='name' />
    </div>
    <div className='part-remark-input-flex' >
      <div className='part-remark-input-text' >备注</div>
      <Input.TextArea onChange={ getInputValue } style={{ width: 300 }}
                      data-path='PART' data-id='remark'
                      rows={ 6 }
                      placeholder='添加备注' ></Input.TextArea>
    </div>
  </div>
  )

export default PartsRemarkInput

//尝试了这种方式，但是不是很好

function SelectOption(props){
  const { arr, keys } = props
  return(
    <Select style={{ width: 300 }} onChange={handleChange}
            value={ changeKeys(arr,keys) }
            Focus = { true }
            showSearch
            optionFilterProp="children"
            filterOption={(input, option) => option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0}
            >
      {
        arr.map((item,index) =>
          <Select.Option key={ item._id } value={ item._id } >
            { item[keys] }
          </Select.Option>)
      }
    </Select>
    )
}

function handleChange(id){
  dispatch('PART_GET_MACHINE_ID', id )
}

function changeKeys(arr,keys){
  const { machineId } = store.getState().part
  if(machineId){
    const array = arr.filter(item => item._id === machineId)
    const obj = array[0]
    return obj[keys]
  }
  else  return null
}
