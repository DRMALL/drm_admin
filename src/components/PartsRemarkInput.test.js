

import React from 'react'
import renderer from 'react-test-renderer'
import PartsRemarkInput from './PartsRemarkInput'

it('PartsRemarkInput test', () => {
  const machineArray = [{ _id: 'sdr34', number: 'a4355', name: '石油设备' },
                        { _id: 'sdr34ds', number: 'a435455', name: '石油设备' }]
  const tree = renderer.create(<PartsRemarkInput machineArray={ machineArray } />)
  expect(tree).toMatchSnapshot()
})
