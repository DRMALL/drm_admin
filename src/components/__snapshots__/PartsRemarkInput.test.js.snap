// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartsRemarkInput test 1`] = `
<div
  className="part-remark-input"
>
  <div
    className="part-remark-input-flex"
  >
    <div
      className="part-remark-input-text"
    >
      设备编号
    </div>
    <div
      className="ant-select ant-select-enabled"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onTouchStart={[Function]}
      style={
        Object {
          "width": 300,
        }
      }
    >
      <div
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup="true"
        className="ant-select-selection
                  ant-select-selection--single"
        onKeyDown={[Function]}
        role="combobox"
        tabIndex={0}
      >
        <div
          className="ant-select-selection__rendered"
        >
          <div
            className="ant-select-selection-selected-value"
            style={
              Object {
                "display": "block",
                "opacity": 1,
              }
            }
            title={null}
          />
          <div
            className="ant-select-search ant-select-search--inline"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <div
              className="ant-select-search__field__wrap"
            >
              <input
                className="ant-select-search__field"
                disabled={undefined}
                onChange={[Function]}
                onKeyDown={[Function]}
                value=""
              />
              <span
                className="ant-select-search__field__mirror"
              >
                
                 
              </span>
            </div>
          </div>
        </div>
        <span
          className="ant-select-arrow"
          onClick={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "userSelect": "none",
            }
          }
          unselectable="unselectable"
        >
          <b />
        </span>
      </div>
    </div>
  </div>
  <div
    className="part-remark-input-flex"
  >
    <div
      className="part-remark-input-text"
    >
      设备名称
    </div>
    <div
      className="ant-select ant-select-enabled"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={undefined}
      onMouseLeave={undefined}
      onTouchStart={[Function]}
      style={
        Object {
          "width": 300,
        }
      }
    >
      <div
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup="true"
        className="ant-select-selection
                  ant-select-selection--single"
        onKeyDown={[Function]}
        role="combobox"
        tabIndex={0}
      >
        <div
          className="ant-select-selection__rendered"
        >
          <div
            className="ant-select-selection-selected-value"
            style={
              Object {
                "display": "block",
                "opacity": 1,
              }
            }
            title={null}
          />
          <div
            className="ant-select-search ant-select-search--inline"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <div
              className="ant-select-search__field__wrap"
            >
              <input
                className="ant-select-search__field"
                disabled={undefined}
                onChange={[Function]}
                onKeyDown={[Function]}
                value=""
              />
              <span
                className="ant-select-search__field__mirror"
              >
                
                 
              </span>
            </div>
          </div>
        </div>
        <span
          className="ant-select-arrow"
          onClick={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "userSelect": "none",
            }
          }
          unselectable="unselectable"
        >
          <b />
        </span>
      </div>
    </div>
  </div>
  <div
    className="part-remark-input-flex"
  >
    <div
      className="part-remark-input-text"
    >
      备注
    </div>
    <textarea
      className="ant-input"
      data-id="remark"
      data-path="PART"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="添加备注"
      rows={6}
      style={
        Object {
          "width": 300,
        }
      }
    />
  </div>
</div>
`;
